pos_customer_display/static/src/js/customer_display.js

```diff
@@ -21,13 +21,26 @@ openerp.pos_customer_display = function(instance){
             var line_length = this.config.customer_display_line_length || 20;
             var currency_rounding = Math.ceil(Math.log(1.0 / this.currency.rounding) / Math.log(10));
 
-            if (type == 'addProduct'){
-                // in order to not recompute qty in options..., we assume that the new ordeLine is the last of the collection
-                // addOrderline exists but is not called by addProduct, should we handle it ?
-                var line = this.get('selectedOrder').getLastOrderline(); 
-                var price_unit = line.get_unit_price() * (1.0 - (line.get_discount() / 100.0));
+            if (type == 'add_update_line'){
+                var line = data['line'];
+                var price_unit = line.get_unit_price();
+                var discount = line.get_discount();
+                if (discount) {
+                    price_unit = price_unit * (1.0 - (discount / 100.0));
+                    }
                 price_unit = price_unit.toFixed(currency_rounding);
-                var l21 = line.get_quantity_str_with_unit() + ' x ' + price_unit;
+                var qty = line.get_quantity();
+                // only display decimals when qty is not an integer
+                if (qty.toFixed(0) == qty) {
+                    qty = qty.toFixed(0);
```

Comment:
I don't get it.

Reply:
The aim of this is to display 2 instead of 2.000, 3 instead of 3.000

Reply:
what i read is : 
`if a == b then 
  b = a
`